calyx-py/calyx/builder.py

```diff
@@ -427,43 +428,64 @@ def binary_use(self, left, right, cell, groupname=None):
             cell.right = right
         return CellAndGroup(cell, comb_group)
 
-    def eq_use(self, left, right, width, signed=False, cellname=None):
+    def try_infer_width(self, width, left, right):
+        """If `width` is None, try to infer it from `left` or `right`.
+        If that fails, raise an error.
+        """
+        width = width or self.infer_width(left) or self.infer_width(right)
+        if not width:
+            raise WidthInferenceError(
+                "Cannot infer widths from `left` or `right`. "
+                "Consider providing width as an argument."
+            )
+        return width
+
+    def eq_use(self, left, right, width=None, signed=False, cellname=None):
         """Inserts wiring into `self` to check if `left` == `right`."""
+        width = self.try_infer_width(width, left, right)
         return self.binary_use(left, right, self.eq(width, cellname, signed))
 
-    def neq_use(self, left, right, width, signed=False, cellname=None):
+    def neq_use(self, left, right, width=None, signed=False, cellname=None):
```

Comment:
I would probably take `width` last since you might want to positionally provide `cellname` without providing a `width` (and same with `signed`).